src/plugins/mpi/pmix/mpi_pmix.c

```diff
@@ -227,6 +293,13 @@ extern mpi_plugin_client_state_t *p_mpi_hook_client_prelaunch(
 		slurm_mutex_unlock(&setup_mutex);
 	}
 
+	slurm_mutex_lock(&abort_mutex);
```

Comment:
Avoid this by 
1. initializing everything as part of **this** thread: read IP/port, open FD.
1. Setting up pmixp Info database fields (need to implement access functions)
1. Accessing FD through pmixp_info* access functions.